### ISSUE TITLE
Release 0.17

### ISSUE THREAD

**Issue by BigLep** [ipfs/kubo#9319]
# Meta
* Release owner: @galargh  
* Release reviewer: @ajnavarro 
* Expected RC date: ~~2022-10-20~~ ~~2022-11-03~~ 2022-11-10
* Expected final release date:  ~~2022-11-03~~ ~~2022-11-10~~ ~~2022-11-17~~ ~~2022-11-21~~ 2022-11-22
* Accompanying PR for improving the release process: https://github.com/ipfs/kubo/pull/9391

See the [Kubo release process](https://pl-strflt.notion.site/Kubo-Release-Process-5a5d066264704009a28a79cff93062c4) for more info.

# Initial planning
## Themes
1. (@guseggert) Filecoin/IPFS interop
   - Reframe in production: https://github.com/ipfs/kubo/issues/9150
   - Reframe GET dag-cbor support
   - Any necessary Bitswap fixes/improvements
2. Prepare for [IPFS Camp](https://2022.ipfs.camp/)
    - Talks
      - (@lidel) Example application based on dag-cbor b-tree
      - (@lidel) Specs
      - (@Jorropo) Data transfer / faster download
      - (@Jorropo) UnixFS
    - General Lisbon support
4. (All) 6-month Roadmapping
5. (@ajnavarro) Protect Kubo nodes by default with the go-libp2p resource manager: https://github.com/ipfs/kubo/issues/9001
7. (@Jorropo) Address DAG handling issues
8. (@Jorropo) UnixFS spec
9. (@hacdias) Gateway:
   - Brave fix around DNSLinks on localhost: https://github.com/ipfs/kubo/issues/9243
   - dag-json / dag-cbor responses: https://github.com/ipfs/kubo/pull/9335
   - TAR responses: https://github.com/ipfs/kubo/pull/9029
7. Cleanup the Kubo RPC story: https://github.com/ipfs/kubo/issues/9142

# Kubo 0.17.0 Release

We're happy to announce Kubo 0.17.0!

As usual, this release includes important fixes, some of which may be critical for security. Unless the fix addresses a bug being exploited in the wild, the fix will _not_ be called out in the release notes. Please make sure to update ASAP. See our [security fix policy](https://github.com/ipfs/go-ipfs/tree/master/docs/releases.md#security-fix-policy) for details.

## 🗺 What's left for release

<List of items with PRs and/or Issues to be considered for this release>

### Required

- [x] https://github.com/ipfs/kubo/pull/9338

### Nice to have

- [x] https://github.com/ipfs/kubo/pull/9387

## 🔦 Highlights

< top highlights for this release notes. For ANY version (final or RCs) >

## ✅ Release Checklist

Checklist:

- [x] **Stage 0 - Prerequisites**
  - [x] Open an issue against [bifrost-infra](https://github.com/protocol/bifrost-infra) ahead of the release ([example](https://github.com/protocol/bifrost-infra/issues/2109)).
    - [x] Spell out all that we want updated - gateways, the bootstraper and the cluster/preload nodes
    - [x] Mention @protocol/bifrost-team in the issue and let them know the expected date of the release
  - [x] Ensure that the `What's left for release` section has all the checkboxes checked. If that's not the case, discuss the open items with Kubo maintainers and update the release schedule accordingly.
  - [x] Create `docs-release-v0.17.0` branch, open a draft PR and keep updating `docs/RELEASE_ISSUE_TEMPLATE.md` on that branch as you go.
  - [x] Ensure you have a [GPG key generated](https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key) and [added to your GitHub account](https://docs.github.com/en/authentication/managing-commit-signature-verification/adding-a-gpg-key-to-your-github-account). This will enable you to created signed tags.
  - [x] Ensure you have [admin access](https://discuss.ipfs.tech/g/admins) to [IPFS Discourse](https://discuss.ipfs.tech/). Admin access is required to globally pin posts. @2color might be able to assist you.
  - [x] Access to [#bifrost](https://filecoinproject.slack.com/archives/C03MMMF606T) channel in FIL Slack might come in handy. Ask the release reviewer to invite you over.
  - [x] Access to [#shared-pl-marketing-requests](https://filecoinproject.slack.com/archives/C018EJ8LWH1) channel in FIL Slack will be required to request social shares. Ask the release reviewer to invite you over.
  - [x] After the release is deployed to our internal infrastructure, you're going to need read access to [IPFS network metrics](https://github.com/protocol/pldw/blob/624f47cf4ec14ad2cec6adf601a9f7b203ef770d/docs/sources/ipfs.md#ipfs-network-metrics) dashboards. Open an access request in https://github.com/protocol/pldw/issues/new/choose if you don't have it yet ([example](https://github.com/protocol/pldw/issues/158)).
  - [x] You're also going to need NPM installed on your system. See [here](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) for instructions.
  - [x] Prepare changelog proposal in [docs/changelogs/v0.17.md](https://github.com/ipfs/kubo/blob/master/docs/changelogs/).
    - Skip filling out the `### Changelog` section (the one where which lists all the commits and contributors) for now. We will populate it after the release branch is cut.
  - [x] Install ZSH ([instructions](https://github.com/ohmyzsh/ohmyzsh/wiki/Installing-ZSH#install-and-set-up-zsh-as-default)). It is needed by the changelog creation script.
  - [x] Ensure you have `kubo` checked out under `$(go env GOPATH)/src/github.com/ipfs/kubo`. This is required by the changelog creation script.
    - If you want your clone to live in a different location, you can symlink it to the expected location by running `mkdir -p $(go env GOPATH)/src/github.com/ipfs && ln -s $(pwd) $(go env GOPATH)/src/github.com/ipfs/kubo`.
  - [x] Ensure that [README.md](https://github.com/ipfs/go-ipfs/tree/master/README.md) is up to date.
- [ ] **Stage 1 - Initial Preparations**
  - [ ] Upgrade to the latest patch release of Go that CircleCI has published (currently used version: `1.19.1`)
    - [x] See the list here: https://hub.docker.com/r/cimg/go/tags
    - [ ] [ipfs/distributions](https://github.com/ipfs/distributions): bump [this version](https://github.com/ipfs/distributions/blob/master/.tool-versions#L2)
    - [ ] [ipfs/kubo](https://github.com/ipfs/kubo): [example PR](https://github.com/ipfs/kubo/pull/8599)
    - [x] [ipfs/ipfs-docs](https://github.com/ipfs/ipfs-docs): [example PR](https://github.com/ipfs/ipfs-docs/pull/1298)
  - [x] Fork a new branch (`release-v0.17.0`) from `master`.
  - [x] Bump the version in `version.go` in the `master` branch to `v0.18.0-dev` via a PR ([example](https://github.com/ipfs/kubo/pull/9305)).
- [x] **Stage 2 - Release Candidate** - _if any [non-trivial](docs/releases.md#footnotes) changes need to be included in the release, return to this stage_
  - [x] Bump the version in `version.go` in the `release-v0.17.0` branch to `v0.17.0-rcN`.
  - [x] If applicable, add new commits to the `release-v0.17.0` branch from `master` using `git cherry-pick -x ...`
      - Note: `release-*` branches are protected. You can do all needed updates on a separated branch (e.g. `wip-release-v0.17.0`) and when everything is settled push to `release-v0.17.0`
  - [x] Push the `release-v0.17.0` branch to GitHub (`git push origin release-v0.17.0`) and create a draft PR targetting `release` branch if it doesn't exist yet ([example](https://github.com/ipfs/kubo/pull/9306)).
  - [x] Wait for CI to run and complete PR checks. All checks should pass.
  - [x] Create a signed tag for the release candidate.
    - [x] This is a dangerous operation, as it is difficult to reverse due to Go modules and automated Docker image publishing. Remember to verify the commands you intend to run for items marked with ⚠️ with the release reviewer.
    - [x] ⚠️ Tag HEAD `release-v0.17.0` commit with `v0.17.0-rcN` (`git tag -s v0.17.0-rcN -m 'Pre-release 0.17.0-rcn'`)
    - [x] Run `git show v0.17.0` to ensure the tag is correct.
    - [x] ⚠️ Push the `v0.17.0` tag to GitHub (`git push origin v0.17.0`; DO NOT USE `git push --tags` because it pushes all your local tags).
  - [x] Add artifacts to https://dist.ipfs.tech by making a PR against [ipfs/distributions](https://github.com/ipfs/distributions)
    - [x] Clone the `ipfs/distributions` repo locally.
    - [x] Create a new branch (`kubo-release-v0.17.0-rcn`) from `master`.
    - [x] Run `./dist.sh add-version kubo v0.17.0-rcN` to add the new version to the `versions` file ([instructions](https://github.com/ipfs/distributions#usage)).
      - `dist.sh` will print _WARNING: not marking pre-release kubo v0.17.0-rc1n as the current version._.
    - [x] Push the `kubo-release-v0.17.0-rcn` branch to GitHub and create a PR from that branch ([example](https://github.com/ipfs/distributions/pull/760)).
    - [x] Ask for a review from the release reviewer.
    - [x] Enable auto-merge for the PR.
      - PR build will build the artifacts and generate a diff in around 30 minutes
      - PR will be merged automatically once the diff is approved
      - `master` build will publish the artifacts to https://dist.ipfs.io in around 30 minutes
    - [x] Ensure that the artifacts are available at https://dist.ipfs.io
  - [x] Publish the RC to [the NPM package](https://www.npmjs.com/package/go-ipfs?activeTab=versions) by running https://github.com/ipfs/npm-go-ipfs/actions/workflows/main.yml (it happens automatically but it is safe to speed up the process and kick of a run manually)
  - [x] Cut a pre-release on [GitHub](https://github.com/ipfs/kubo/releases) ([instructions](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release), [example](https://github.com/ipfs/kubo/releases/tag/v0.16.0-rc1))
    - Use `v0.17.0-rcN` as the tag.
    - Link to the release issue in the description.
    - Link to the relevant [changelog](https://github.com/ipfs/kubo/blob/master/docs/changelogs/) in the description.
    - Check `This is a pre-release`.
  - [x] Synchronize release artifacts by running [sync-release-assets](https://github.com/ipfs/kubo/actions/workflows/sync-release-assets.yml) workflow.
  - [x] Announce the RC
    - [x] Create a new post on [IPFS Discourse](https://discuss.ipfs.tech). ([example](https://discuss.ipfs.tech/t/kubo-v0-16-0-rc1-release-candidate-is-out/15248))
      - Use `Kubo v0.17.0-rcn Release Candidate is out!` as the title.
      - Use `kubo` and `go-ipfs` as topics.
      - Repeat the title as a heading (`##`) in the description.
      - Link to the GitHub Release, binaries on IPNS, docker pull command and release notes in the description.
    - [x] Pin the topic globally so that it stays at the top of the category.
    - [x] Check if Discourse post was automatically copied to:
      - [x] IPFS Discord #ipfs-chatter
      - [x] FIL Slack #ipfs-chatter
      - [x] Matrix https://matrix.to/#/#ipfs-chatter:ipfs.io
    - [x] Mention [early testers](https://github.com/ipfs/go-ipfs/tree/master/docs/EARLY_TESTERS.md) in the comment under the release issue ([example](https://github.com/ipfs/kubo/issues/9237#issuecomment-1258072509)).
- [x] **Stage 3 - Internal Testing**
  - [x] Infrastructure Testing.
    - [x] Update the issue against [bifrost-infra](https://github.com/protocol/bifrost-infra) ([example](https://github.com/protocol/bifrost-infra/issues/2109)).
      - [x] Mention @protocol/bifrost-team in the issue to let them know the release is ready
      - [x] [Optional] Reply under a message about the issue in the #bifrost channel on FIL Slack once the RC is out. Send the message to the channel.
    - [x] Check [metrics](https://protocollabs.grafana.net/d/8zlhkKTZk/gateway-slis-precomputed?orgId=1) every day.
      - Compare the metrics trends week over week.
      - If there is an unexpected variation in the trend, message the #bifrost channel on FIL Slack and ask for help investigation the cause.
  - [x] IPFS Application Testing.
    - [x] [IPFS Desktop](https://github.com/ipfs-shipyard/ipfs-desktop)
      - [x] Upgrade to the RC in [ipfs-desktop](https://github.com/ipfs-shipyard/ipfs-desktop)
      - [x] Run `npm install` to update `package-lock.json`.
      - [x] Push to a branch ([example](https://github.com/ipfs/ipfs-desktop/pull/1826/commits/b0a23db31ce942b46d95965ee6fe770fb24d6bde))
      - [x] Open a draft PR to track through the final release ([example](https://github.com/ipfs/ipfs-desktop/pull/1826))
      - [x] Ensure CI tests pass
    - [x] [IPFS Companion](https://github.com/ipfs-shipyard/ipfs-companion)
      - [x] Start kubo daemon of the version to release.
      - [x] Start a fresh chromium or chrome instance using `chromium --user-data-dir=$(mktemp -d)` (macos `/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome --user-data-dir=$(mktemp -d)`)
      - [x] Start a fresh firefox instance using `firefox --profile $(mktemp -d)` (macos `/Applications/Firefox.app/Contents/MacOS/firefox --profile $(mktemp -d)`)
      - [x] Install IPFS Companion from [vendor-specific store](https://github.com/ipfs/ipfs-companion/#readme).
      - [x] Check that the comunication between Kubo daemon and IPFS companion is working properly checking if the number of connected peers changes.
- [x] **Stage 5 - Release** - _ONLY FOR FINAL RELEASE_
  - [x] Prepare the `release` branch.
    - [x] Bump the version in `version.go` in the `release-v0.17.0` branch to `v0.17.0`.
    - [x] Update the [docs/changelogs/v0.17.md](docs/changelogs) with the new commits and contributors.
      - [x] Run `./bin/mkreleaselog` twice to generate the changelog and copy the output.
      - The first run of the script might be poluted with `git clone` output.
      - [x] Paste the output into the `### Changelog` section of the changelog file inside the `<details><summary></summary></details>` block.
      - [x] Commit the changelog changes.
    - [x] Push the `release-v0.17.0` branch to GitHub (`git push origin release-v0.17.0`)
    - [x] Mark the PR created from `release-v0.17.0` as ready for review.
      - [x] Ensure the PR is targetting `release` branch.
      - [x] Ensure that CI is green.
      - [x] Have release reviewer review the PR.
    - [x] Merge the PR into `release` branch using the `Create a merge commit` (do **NOT** use `Squash and merge` nor `Rebase and merge` because we need to be able to sign the merge commit).
      - Do not delete the `release-v0.17.0` branch.
    - [x] Checkout the `release` branch locally.
      - Remember to pull the latest changes.
    - [x] Create a signed tag for the release.
      - [x] This is a dangerous operation, as it is difficult to reverse due to Go modules and automated Docker image publishing. Remember to verify the commands you intend to run for items marked with ⚠️ with the release reviewer.
      - [x] ⚠️ Tag HEAD `release` commit with `v0.17.0` (`git tag -s v0.17.0 -m 'Release 0.17.0'`)
      - [x] Run `git show v0.17.0` to ensure the tag is correct.
      - [x] ⚠️ Push the `v0.17.0` tag to GitHub (`git push origin v0.17.0`; DO NOT USE `git push --tags` because it pushes all your local tags).
  - [ ] Publish the release.
    - [x] Wait for [Publish docker image](https://github.com/ipfs/kubo/actions/workflows/docker-image.yml) workflow run initiated by the tag push to finish.
    - [x] Add artifacts to https://dist.ipfs.tech by making a PR against [ipfs/distributions](https://github.com/ipfs/distributions)
      - [x] Clone the `ipfs/distributions` repo locally.
      - [x] Create a new branch (`kubo-release-v0.17.0`) from `master`.
      - [x] Run `./dist.sh add-version kubo v0.17.0` to add the new version to the `versions` file ([instructions](https://github.com/ipfs/distributions#usage)).
      - [x] Push the `kubo-release-v0.17.0` branch to GitHub and create a PR from that branch ([example](https://github.com/ipfs/distributions/pull/768)).
      - [x] Ask for a review from the release reviewer.
      - [x] Enable auto-merge for the PR.
        - PR build will build the artifacts and generate a diff in around 30 minutes
        - PR will be merged automatically once the diff is approved
        - `master` build will publish the artifacts to https://dist.ipfs.io in around 30 minutes
      - [x] Ensure that the artifacts are available at https://dist.ipfs.io
    - [x] Publish the release to [the NPM package](https://www.npmjs.com/package/go-ipfs?activeTab=versions) by running https://github.com/ipfs/npm-go-ipfs/actions/workflows/main.yml (it happens automatically but it is safe to speed up the process and kick of a run manually)
  - [x] Cut the release on [GitHub](https://github.com/ipfs/kubo/releases) ([instructions](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release), [example](https://github.com/ipfs/kubo/releases/tag/v0.16.0))
    - Use `v0.17.0` as the tag.
    - Link to the release issue in the description.
    - Copy the relevant [changelog](https://github.com/ipfs/kubo/blob/release/docs/changelogs/) into the release description.
      - Keep the release notes as trim as possible (e.g. remove top headers where possible, [example](https://github.com/ipfs/kubo/releases/tag/v0.15.0))
  - [x] Synchronize release artifacts by running [sync-release-assets](https://github.com/ipfs/kubo/actions/workflows/sync-release-assets.yml) workflow.
  - [x] Announce the release
    - [x] Add a link to the release to this release issue as a comment.
    - [x] Create a new post on [IPFS Discourse](https://discuss.ipfs.tech). ([example](https://discuss.ipfs.tech/t/kubo-v0-16-0-release-is-out/15286))
      - Use `Kubo v0.17.0 Release is out!` as the title.
      - Use `kubo` and `go-ipfs` as topics.
      - Repeat the title as a heading (`##`) in the description.
      - Link to the GitHub Release, binaries on IPNS, docker pull command and release notes in the description.
    - [x] Pin the topic globally so that it stays at the top of the category.
    - [x] Check if Discourse post was automatically copied to:
      - [x] IPFS Discord #ipfs-chatter
      - [x] FIL Slack #ipfs-chatter
      - [x] Matrix
  - [x] Add a link from release notes to Discuss post (like we did here: https://github.com/ipfs/kubo/releases/tag/v0.15.0)
  - [x] Update the draft PR created for [interop](https://github.com/ipfs/interop) to use the new release and mark it as ready for review.
  - [x] Update the draft PR created for [IPFS Desktop](https://github.com/ipfs-shipyard/ipfs-desktop) to use the new release and mark it as ready for review.
  - [x] Update docs
    - [x] Run https://github.com/ipfs/ipfs-docs/actions/workflows/update-on-new-ipfs-tag.yml to generate a PR to the docs repo
    - [x] Merge the auto-created PR in https://github.com/ipfs/ipfs-docs/pulls ([example](https://github.com/ipfs/ipfs-docs/pull/1263))
  - [x] Get the blog post created and shared
    - [x] Submit a request for blog post creation using [the form](https://airtable.com/shrNH8YWole1xc70I).
      - Notify marketing in #shared-pl-marketing-requests about the blog entry request (since the form tends to go to spam; [example]([example](https://filecoinproject.slack.com/archives/C018EJ8LWH1/p1664885305374909))).
    - [x] The post is live on https://blog.ipfs.io
    - [x] Share the blog post
      - [x] Twitter (request in Filecoin Slack channel #shared-pl-marketing-requests; [example](https://filecoinproject.slack.com/archives/C018EJ8LWH1/p1664903524843269?thread_ts=1664885305.374909&cid=C018EJ8LWH1))
      - [x] [Reddit](https://reddit.com/r/ipfs)
- [x] **Stage 6 - Post-Release**
  - [x] Merge the `release` branch back into `master`, ignoring the changes to `version.go` (keep the `-dev` version from master).
  - [x] Create an issue using this release issue template for the _next_ release.
  - [x] Close this release issue.

## How to contribute?

Would you like to contribute to the IPFS project and don't know how? Well, there are a few places you can get started:

- Check the issues with the `help wanted` label in the [ipfs/kubo repo](https://github.com/ipfs/kubo/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22)
- Join an IPFS All Hands, introduce yourself and let us know where you would like to contribute - https://github.com/ipfs/team-mgmt/#weekly-ipfs-all-hands
- Hack with IPFS and show us what you made! The All Hands call is also the perfect venue for demos, join in and show us what you built
- Join the discussion at [discuss.ipfs.io](https://discuss.ipfs.io/) and help users finding their answers.
- Join the [🚀 IPFS Core Implementations Weekly Sync 🛰](https://github.com/ipfs/team-mgmt/issues/992) and be part of the action!

**Comment by BigLep** [comment]
@galargh : can you please followup with the team to see if either of these need to be addressed before we start the release:
1. https://github.com/ipfs/kubo/issues/9374
2. https://github.com/ipfs/kubo/issues/8585

**Comment by p-shahi** [comment]
> 0.16.0 is crashing

Started a discussion in the linked libp2p issue. Please let us know if you need a fix in 0.17.0


A separate question: will Kubo update to https://github.com/libp2p/go-libp2p/releases/tag/v0.23.3 to resolve https://github.com/libp2p/go-libp2p-relay-daemon/issues/18#issuecomment-1276212041?

**Comment by ajnavarro** [comment]
@p-shahi Update here #9387

**Comment by BigLep** [comment]
2022-11-08 maintainer: conversation on what's needed before RC:
- [ ] @guseggert Updates config.md and release notes around Reframe in light of whatever gets decided for Reframe/HTTP Delegated routing in https://github.com/ipfs/specs/pull/337
- [ ] go-libp2p update: https://github.com/ipfs/kubo/pull/9387 (@guseggert to review)

These are nice to haves that we'd like to see land in time:
- [ ] @lidel New WebUI release with updated GeoIP DB and WebTransport fix
- [ ] @ajnavarro Resource Manager on by default:
  - [ ] https://github.com/ipfs/kubo/pull/9338
  - [ ] https://github.com/ipfs/kubo/issues/9385
  - [ ] https://github.com/ipfs/kubo/pull/9351

Items that will get fixed in the RC phase or in a followup release:
1. https://github.com/ipfs/kubo/issues/9374
2. https://github.com/ipfs/kubo/issues/8585

The plan regardless is still to do an RC on 2022-11-10.

**Comment by p-shahi** [comment]
About to release 0.23.4 momentarily with some more changes, so let's hold off on merging this and wait for that.

**Comment by galargh** [comment]
2022-11-10: Investigating 2 issues that are currently halting the release:
- [ ] webui tests failing for https://github.com/ipfs/kubo/pull/9392
- [ ] collab cluster being unreachable during builds of https://github.com/ipfs/distributions/pull/773

**Comment by BigLep** [comment]
2022-11-10 conversation:
1. We'll put the go upgrade on the shelf
2. Goal is to get through Stage 2 today (if collab cluster is operational again)
3. @guseggert will get in resource manager notes into https://github.com/ipfs/kubo/blob/master/docs/changelogs/v0.17.md

**Comment by galargh** [comment]
Early testers ping for RC1 testing 😄

- [ ]  pacman.store (@RubenKelevra)
- [ ]  Infura (@MichaelMure)
- [ ]  Textile (@sanderpick)
- [ ]  Pinata (@obo20)
- [ ]  RTrade (@postables)
- [ ]  QRI (@b5)
- [ ]  Siderus (@koalalorenzo)
- [ ]  Charity Engine (@rytiss, @tristanolive)
- [ ]  Fission (@bmann)
- [ ]  OrbitDB (@tabcat)

**Comment by BigLep** [comment]
@lidel @galargh : I think we should put a proactive message in #ipfs-operators about this RC given the criticality of resource manager being enabled by default and that being important for their node safety.  I'm happy to sync up offline regarding messaging.

**Comment by BigLep** [comment]
2022-11-15 conversation:
We're going to make to a new RC that will include:
1. #9407
2. https://github.com/ipfs/kubo/pull/8913
3. https://github.com/ipfs/kubo/pull/9402

Seperately we'll look at Bifrost resource manager default setting: https://github.com/protocol/bifrost-infra/blob/master/ansible/inventories/bifrost/group_vars/gateway_ipfs_bank1.yml

**Comment by lidel** [comment]
@galargh  FYSA I've reviewed/merged/cherry-picked remaining things, should be ready for  RC2 
(see the tail of the `release-v0.17.0` changes in your PR https://github.com/ipfs/kubo/pull/9395).

(missing changelog entries can be updated later, it does not  block  RC2 imo)

**Comment by galargh** [comment]
> @galargh FYSA I've reviewed/merged/cherry-picked remaining things, should be ready for RC2 (see the tail of the `release-v0.17.0` changes in your PR #9395).
> 
> (missing changelog entries can be updated later, it does not block RC2 imo)

Thank you! I'll be starting the process for RC2 release shortly.

**Comment by BigLep** [comment]
@galargh : There are various doc updates (including changelog in https://github.com/ipfs/kubo/pull/9413 ).  I guess since these are doc based, they don't have to make it in the RC.

When Bifrost attempts to deploy this RC-2 out, I believe we should have them try simplified configuration per https://github.com/protocol/bifrost-infra/issues/1815#issuecomment-1316651157 so that we're actually getting vetting of the default configuration.

**Comment by galargh** [comment]
Early testers ping for RC2 testing 😄. 

- [ ]  pacman.store (@RubenKelevra)
- [ ]  Infura (@MichaelMure)
- [ ]  Textile (@sanderpick)
- [ ]  Pinata (@obo20)
- [ ]  RTrade (@postables)
- [ ]  QRI (@b5)
- [ ]  Siderus (@koalalorenzo)
- [ ]  Charity Engine (@rytiss, @tristanolive)
- [ ]  Fission (@bmann)
- [ ]  OrbitDB (@tabcat)

You're getting this message because you're listed [here](https://github.com/ipfs/kubo/blob/master/docs/EARLY_TESTERS.md#who-has-signed-up).  Please update this list if you no longer want to be included.

**Comment by galargh** [comment]
Given we released the RC2 yesterday and are still in the process of rolling it out to internal infrastructure, we're most likely going to postpone the final release until next week. To be confirmed later today.

**Comment by galargh** [comment]
Moving the final release to 2022-11-21.

**Comment by galargh** [comment]
The final release is here! https://github.com/ipfs/kubo/releases/tag/v0.17.0

**Comment by galargh** [comment]
Next release -> https://github.com/ipfs/kubo/issues/9417
